### PR TITLE
Migrate core to TS p3

### PR DESCRIPTION
**Pull request by albertogasparin**
Another batch of `loot-core` migrated.

### CODE REVIEW

File: packages/loot-core/src/client/actions/account.ts
Comment (albertogasparin) [inline packages/loot-core/src/client/actions/account.ts:109]: This `send` API would need a much better structure to become "type safe", but that will be a solid undertake on its own, so for now we just require some types at call site.

File: packages/loot-core/src/server/accounts/sync.ts
Comment (albertogasparin) [inline packages/loot-core/src/server/accounts/sync.ts:215]: This is a runtime changes as I've found a mismatch: consumers expect `accountBalance` but we return `accountBalances`. I picked one 🤷

File: packages/loot-core/src/server/budget/base.ts
Comment (albertogasparin) [inline packages/loot-core/src/server/budget/base.ts:98]: `sheetForMonth` does not take a second parameter

File: packages/loot-core/src/server/crdt/timestamp.ts
Comment (albertogasparin) [inline packages/loot-core/src/server/crdt/timestamp.ts:333]: I assumed we want to customise error name, not add a new custom property called type

File: packages/loot-core/src/server/sync/encoder.ts
Comment (albertogasparin) [inline packages/loot-core/src/server/sync/encoder.ts:5]: `sync_pb` is too much of a beast to covert it to ESM...

Comment (sinistersnare) [reply packages/loot-core/src/server/sync/encoder.ts:5]: Can't you just regenerate `sync_pb` by recompiling `sync.proto`?

File: packages/loot-core/src/client/actions/account.ts
Comment (j-f1) [reply packages/loot-core/src/client/actions/account.ts:109]: Would it be possible to declare a type like...

```ts
interface Handlers {
  'nordigen-accounts-sync': ({ id }) => Promise<void>;
  // TODO: remove once all handlers are typed
  [key: string]: (...args: unknown[]) => Promise<unknown>;
}
```

File: packages/loot-core/src/server/sync/proto/sync_pb.js
Comment (j-f1) [inline packages/loot-core/src/server/sync/proto/sync_pb.js]: I believe this was explicitly changed at one point to remove the usage of eval (`[new] Function('return this')()`)

File: packages/loot-core/src/client/actions/account.ts
Comment (albertogasparin) [inline packages/loot-core/src/client/actions/account.ts:231]: Now that `send` is typed, could not find `banks` handler and seems like this is not used anymore

File: packages/loot-core/src/server/accounts/qif2json.ts
Comment (j-f1) [inline packages/loot-core/src/server/accounts/qif2json.ts:8]: Maybe prefix these names to clarify that they aren’t Actual transactions/divisions?
```suggestion
type QIFTransaction = {
```

File: packages/loot-core/src/server/accounts/sync.test.ts
Comment (j-f1) [inline packages/loot-core/src/server/accounts/sync.test.ts:316]: Would this be flaky if the code changes to call `currentDay()` more than once?

File: packages/loot-core/src/client/data-hooks/payees.tsx
Comment (j-f1) [inline packages/loot-core/src/client/data-hooks/payees.tsx:37]: Why `= undefined` instead of declaring the parameter optional with types?

File: packages/loot-core/src/server/accounts/transfer.test.ts
Comment (albertogasparin) [reply packages/loot-core/src/server/accounts/transfer.test.ts:37]: FYI I've created a `types` in `loot-core/src`

File: packages/loot-core/src/server/accounts/sync.ts
Comment (j-f1) [reply packages/loot-core/src/server/accounts/sync.ts:215]: fyi @fstybel — you touched this last, wanted to make sure this change is as expected.

File: packages/loot-core/src/client/actions/account.ts
Comment (j-f1) [reply packages/loot-core/src/client/actions/account.ts]: Super nice, thanks for making that so through!

File: packages/loot-core/src/server/accounts/transfer.test.ts
Comment (j-f1) [reply packages/loot-core/src/server/accounts/transfer.test.ts:37]: If you want to share types between this repo and `actual-server` you can export them from `@actual-app/api` which is imported by the server.

File: packages/loot-core/src/server/budget/base.ts
Comment (j-f1) [reply packages/loot-core/src/server/budget/base.ts:98]: This is a bit concerning to me since there are actually two different budget types and presumably the point was to grab the appropriate one without them clobbering each other (but I haven’t seen any issues with this. @jlongster do you have any more context here?

File: packages/loot-core/src/server/crdt/timestamp.ts
Comment (j-f1) [reply packages/loot-core/src/server/crdt/timestamp.ts:333]: Makes sense!

File: packages/loot-core/src/server/accounts/transfer.test.ts
Comment (albertogasparin) [reply packages/loot-core/src/server/accounts/transfer.test.ts:37]: We can always export a subset there, as that gets build so it is likely easy to bundle them. I was more specifically  asking about `server`/`client`/... folders in `loot-core`

**Review by j-f1 — APPROVED** [review]
Overall LGTM but I would ideally like to hear back from James & Filip before releasing these changes. 

Since we are likely to do a bug fix release in the very near future, I would like to hold off merging for a bit but I also don’t want to slow down your migration work too much.

**Comment by albertogasparin** [comment]
What do we want to do? I really wanna merge this and keep migrating the rest of the codebase. So do you prefer me revert the `accountBalance` change and add a TS suppression or merge this anyway (after fixing conflicts)?

**Comment by j-f1** [comment]
Since it’s been almost a week with no reply, I’m ok with merging for now and making changes in the future if either one turns out to be a problem.

**Review by j-f1 — APPROVED** [review]
Thanks!

**Comment by albertogasparin** [comment]
New ofx code returned different shape for `error`, so had to make TS happy again. One more reason we wanna speed up conversion as interfaces become inconsistent too easily

**Comment by j-f1** [comment]
Yep!

**Comment by j-f1** [comment]
Thanks for your work on this :)

**Comment by sinistersnare** [comment]
Thanks for integrating my changes to the Libofx interface!!

**Pull request merged**
